src/ResourceManager/Network/Commands.Network/VirtualNetworkGateway/NewAzureVpnClientConfigurationCommand.cs

```diff
@@ -0,0 +1,152 @@
+                MNM.ProcessorArchitecture.Amd64.ToString() :
+                this.ProcessorArchitecture;
+
+            vpnClientParams.AuthenticationMethod = string.IsNullOrWhiteSpace(this.AuthenticationMethod)
+                ? MNM.AuthenticationMethod.EAPTLS.ToString()
+                : this.AuthenticationMethod;
+
+            // Read the radius server root certificate if present
+            if (!string.IsNullOrWhiteSpace(RadiusRootCert))
+            {
+                if (File.Exists(RadiusRootCert))
+                {
+                    try
+                    {
+                        X509Certificate2 radiusRootCertificate = new X509Certificate2(RadiusRootCert);
+                        vpnClientParams.RadiusServerAuthCertificate = Convert.ToBase64String(radiusRootCertificate.Export(X509ContentType.Cert));
+                    }
+                    catch (Exception)
+                    {
+                        WriteWarning("Invalid radius root certificate specified at path " +
+                                     RadiusRootCert);
+                    }
+                }
+                else
+                {
+                    WriteWarning("Cannot find radius root certificate with path " +
+                                 RadiusRootCert);
+                }
+            }
+
+            // Read the radius server root certificate if present
+            if (!string.IsNullOrWhiteSpace(this.ClientRootCertificates))
+            {
+                foreach (string clientRootCertPath in this.ClientRootCertificates.Split(','))
+                {
+                    vpnClientParams.ClientRootCertificates = new List<string>();
+                    if (File.Exists(clientRootCertPath))
+                    {
+                        try
+                        {
+                            X509Certificate2 clientRootCertificate = new X509Certificate2(clientRootCertPath);
+                            vpnClientParams.ClientRootCertificates.Add(
+                                Convert.ToBase64String(clientRootCertificate.Export(X509ContentType.Cert)));
+                        }
+                        catch (Exception)
+                        {
+                            WriteWarning("Invalid cer file specified for client root certificate with path " +
+                                         clientRootCertPath);
+                        }
+                    }
+                    else
+                    {
+                        WriteWarning("Cannot find client root certificate with path " +
+                                     clientRootCertPath);
+                    }
+                }
+            }
+
+            var vnetVpnClientParametersModel = Mapper.Map<MNM.VpnClientParameters>(vpnClientParams);
+
+            //TODO:- This code is added just for current release of P2S feature as Generatevpnclientpackage API is broken & need to be fixed on server 
```

Comment:
Copy paste from existing commandlet that does something similar. This is a new commandlet that exposes similar functionality to the existing Get-AzureVpnClientPackage command in addition to some new options that we have exposed.